### PR TITLE
618 add how to section to documentation

### LINKED ISSUE #618: Add How to Section to Documentation
We have discussed many times that the current documentation is good as a reference when you know what you are looking for - but it it's not so good at explaining how to do specific high-level things that might involve multiple elements.

For example, how to change the solver (https://github.com/spine-tools/SpineOpt.jl/issues/616) or how to define an efficiency (https://github.com/spine-tools/SpineOpt.jl/discussions/617).

I was thinking that we could add a "How To" section to the documentation and add articles which in many cases the content may exist already in issues or discussions. Ideally, it would be pretty easy to add these articles so they can be easily added to over time. For example, if we could have a how to folder and just drop in MD files of the relevant how_to article and the contents/index could be generated automatically based on the contents of the How To folder.

The two issues/discussions above would be a great start.

I have added an issue label "Add how to article to docs" so we can tag issues that may have useful content for a how-to article.

### PR DESCRIPTION
**Pull request by tarskul**
Added new chapters:
+ How To
+ Implementation detail

Also there are some minor details that needed to be changed to better accommodate the tutorial chapter.

This pull request might be unnecessary as the pull request draganddrop contains the same changes and the draganddrop feature. They are only made separate in case the new chapters are accepted but the draganddrop feature is not.

Fixes #618

## Checklist before merging
- [x] Documentation is up-to-date
- [x] Unit tests have been added/updated accordingly
- [x] Code has been formatted according to SpineOpt's style
- [x] Unit tests pass

### CODE REVIEW

File: docs/src/howto/change_the_solver.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+![image](../figs/objectparametervalue_solveroptions.png)
+
+This is what the solver options map parameter value looks like:
+
+![image](../figs/map_solveroptions.png)
+
+To get a head start with solver options, you can copy their default map values from the parameter definition tab like this:
+
+![image](../figs/objectparameterdefinition_solveroptions.png)
```
Comment (Tasqu) [inline docs/src/howto/change_the_solver.md:20]: Image missing. They need to be included (one way or another) for the final built documentation to display them. Not sure why .png files are ignored, even though the documentation includes .png files :/

File: docs/src/howto/define_an_efficiency.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+- In addition to `fix_ratio_out_in_unit_flow` you have \[constraint\]\_ratio\_\[direction1\]\_\[direction2\]\_unit\_flow where constraint can be min, max or fix and determines the sense of the constraint (max: <, min: >, fix: =) while direction1 and direction2 are used to interpret the direction of the flows involved. `In` signifies an input flow to the unit while `out` signifies an output flow from the unit. For each of these parameters, there is a corresponding \[constraint\]\_\[direction1\]\_\[direction2\]\_units\_on\_coefficient. For example: `max_ratio_in_out_unit_flow` creates the following constraint:
+`input_flow` < `max_ratio_in_out_unit_flow` * `output_flow` + `max_units_on_coefficient_in_out` * `units_on`
+
+## real world example: Compressed Air Energy Storage
+
+To give a feeling for why these functionalities are useful, consider the following real world example for Compressed Air Energy Storage:
+
+![image](../figs/definingconversionsforunits_realworldexample.png)
```
Comment (Tasqu) [inline docs/src/howto/define_an_efficiency.md:30]: Images missing.

**Review by Tasqu — CHANGES_REQUESTED** [review]
Other than the missing image files, this seems ok to merge for me. The `master` branch seems to have issues of its own with missing documentation and broken cross-references, but they don't seem related to these changes as far as I can tell.

**Comment by tarskul** [comment]
Closing the branch in favour of the draganddrop branch #693

**Pull request closed without merging**